src/pages/nft/claim/index.vue

```diff
@@ -1223,14 +1230,47 @@ export default {
         );
       }
     },
-    handleViewCollection() {
+    async ensureClassIdInCollected() {
+      this.isViewCollectionLoading = true;
+
+      let collectedNFTs = this.getCollectedNFTClassesByAddress(
+        this.claimingAddress
+      );
+      let attempts = 0;
+
+      while (
+        !collectedNFTs?.some(item => item.classId === this.classId) &&
+        attempts < MAX_ATTEMPT
+      ) {
+        // eslint-disable-next-line no-await-in-loop
+        await this.fetchCollectedNFTClassesByAddress({
+          address: this.claimingAddress,
+          nocache: true,
+        });
+
+        collectedNFTs = this.getCollectedNFTClassesByAddress(
+          this.claimingAddress
+        );
+
+        attempts += 1;
+
+        // eslint-disable-next-line no-await-in-loop
+        await new Promise(resolve => setTimeout(resolve, 500));
```

Comment:
We have a sleep util
https://github.com/likecoin/liker-land/blob/4d007cd7731600803f9da21371209ef14fffff32/src/util/misc.js#L1-L3